test/vm/untyped_ir_interpreter/instruction_table.hpp

```diff
@@ -0,0 +1,1951 @@
+    {
+        check_requirements<JUMPI, traits>(
+            ctx, stack_bottom, stack_top, gas_remaining);
+        auto const &target = pop(stack_top);
+        auto const &cond = pop(stack_top);
+
+        if (cond) {
+            if (auto it = analysis.jump_coerce_map.find(
+                    static_cast<std::size_t>(prev_ptr - analysis.code()));
+                it != analysis.jump_coerce_map.end()) {
+                cast_pointers(analysis, stack_top, it->second);
+            }
+
+            auto const *const new_ip = [&]() {
+                if (auto it = analysis.jumpdest_addr.find(
+                        static_cast<std::size_t>(prev_ptr - analysis.code()));
+                    it != analysis.jumpdest_addr.end()) {
+                    return reinterpret_cast<std::uint8_t const *>(
+                        static_cast<uintptr_t>(target[0]));
+                }
+                else {
+                    return jump_impl(analysis, target);
+                }
+            }();
+
+            if (MONAD_VM_UNLIKELY(!new_ip)) {
+                ctx.exit(Error);
+            }
+
+            MONAD_VM_MUST_TAIL return instruction_table_untyped_ir<
+                traits>[*new_ip](
+                ctx,
+                analysis,
+                stack_bottom,
+                stack_top,
+                gas_remaining,
+                nullptr,
+                new_ip);
+        }
+        else {
+            if (auto it = analysis.fallthrough_coerce_map.find(
+                    static_cast<std::size_t>(prev_ptr - analysis.code()));
+                it != analysis.fallthrough_coerce_map.end()) {
+                cast_pointers(analysis, stack_top, it->second);
+            }
+            ++instr_ptr;
+            MONAD_VM_MUST_TAIL return instruction_table_untyped_ir<
+                traits>[*instr_ptr](
+                ctx,
+                analysis,
+                stack_bottom,
+                stack_top,
+                gas_remaining,
+                nullptr,
+                instr_ptr);
+        }
+    }
+
+    template <Traits traits>
+    MONAD_VM_INSTRUCTION_CALL void jumpdest(
+        runtime::Context &ctx, IntercodeUntypedIR const &analysis,
+        runtime::uint256_t const *stack_bottom, runtime::uint256_t *stack_top,
+        std::int64_t gas_remaining, std::uint8_t const *prev_ptr,
+        std::uint8_t const *instr_ptr)
+    {
+        fuzz_tstore_stack(
+            ctx,
+            stack_bottom,
+            stack_top,
+            static_cast<uint64_t>(instr_ptr - analysis.code()));
+        check_requirements<JUMPDEST, traits>(
+            ctx, stack_bottom, stack_top, gas_remaining);
+
+        if (prev_ptr) {
+            if (auto it = analysis.fallthrough_coerce_map.find(
+                    static_cast<std::size_t>(prev_ptr - analysis.code()));
+                it != analysis.fallthrough_coerce_map.end()) {
+                cast_pointers(analysis, stack_top, it->second);
+            }
+        }
```

Comment:
This is where we perform the casting of stack values to pointers based on the untyped IR, but only in the fallthrough case, indicated by prev_ptr not being null. 
If prev_ptr is null, either we are at the entry-point, which does not have any coersions as the stack is empty, or the jumpdest was reached via a jump instruction, in which case the coercions have already been performed.